addons/product/pricelist.py

```diff
@@ -252,86 +178,72 @@ def _price_rule_get_multi(self, cr, uid, pricelist, products_by_qty_by_partner,
                     # Ignored - incompatible UoM in context, use default product UoM
                     pass
 
-            price_type = 'standard_price' if pricelist.type == 'purchase' else 'list_price'
+            price_type = 'list_price'
             # if Public user try to access standard price from website sale, need to call _price_get.
             price = product_obj._price_get(cr, uid, [product], price_type, context=context)[product.id] or False
             for rule in items:
-                if rule.min_quantity and qty_in_product_uom < rule.min_quantity:
-                    continue
-                if is_product_template:
-                    if rule.product_tmpl_id and product.id != rule.product_tmpl_id.id:
-                        continue
-                    if rule.product_id and \
-                            (product.product_variant_count > 1 or product.product_variant_ids[0].id != rule.product_id.id):
-                        # product rule acceptable on template if has only one variant
-                        continue
-                else:
-                    if rule.product_tmpl_id and product.product_tmpl_id.id != rule.product_tmpl_id.id:
-                        continue
-                    if rule.product_id and product.id != rule.product_id.id:
+                if ((rule.date_start is False) or (rule.date_start <= date)) and ((rule.date_end is False) or (rule.date_end >= date)):
```

Comment:
better performance: you should put all rule conditions in sql query.

so move condition into sql query from line : https://github.com/odoo-dev/odoo/blob/master-simplify-pricelists-rmu/addons/product/pricelist.py#L185 to line : https://github.com/odoo-dev/odoo/blob/master-simplify-pricelists-rmu/addons/product/pricelist.py#L200